chains/evm/calls/bridge.go

```diff
@@ -226,6 +228,26 @@ func PrepareVoteProposalInput(sourceDomainID uint8, depositNonce uint64, resourc
 	return input, nil
 }
 
+func SimulateVoteProposal(evmCaller ContractCallerClient, prop *proposal.Proposal, tries int, maxTries int, shouldVoteCheckPeriod int) error {
+	input, err := PrepareVoteProposalInput(prop.Source, prop.DepositNonce, prop.ResourceId, prop.Data)
+	if err != nil {
+		return err
+	}
+	msg := ethereum.CallMsg{From: common.Address{}, To: &prop.BridgeAddress, Data: input}
+
+	_, err = evmCaller.CallContract(context.TODO(), ToCallArg(msg), nil)
+	if err != nil {
+		if tries < maxTries {
```

Comment:
This logic should be in voter I think.
Just the simulate vote proposal should be here.

Reply:
To explain more clearly -----> call should be in calls, but the actual logic of retrying should be in voter IMO